File: OJ practices/LeetCode题目集/674. Longest Continuous Increasing Subsequence(easy).java
/*
 * @Autor: SourDumplings
 * @Date: 2019-10-14 08:02:51
 * @Link: https://github.com/SourDumplings/
 * @Email: changzheng300@foxmail.com
 * @Description: https://leetcode.com/problems/longest-continuous-increasing-subsequence/
 */

class Solution
{
    public int findLengthOfLCIS(int[] nums)
    {
        int n = nums.length;
        if (n == 0)
        {
            return 0;
        }
        int res = 0, count = 1;
        for (int i = 1; i < n; i++)
        {
            if (nums[i] <= nums[i - 1])
            {
                if (count > res)
                {
                    res = count;
                }
                count = 1;
            }
            else
            {
                ++count;
            }
        }
        if (count > res)
        {
            res = count;
        }
        return res;
    }
}

File: OJ practices/LeetCode题目集/674. Longest Continuous Increasing Subsequence(easy).cpp
/*
 * @Author: SourDumplings
 * @Date: 2019-10-14 07:53:50
 * @Link: https://github.com/SourDumplings/
 * @Email: changzheng300@foxmail.com
 * @Description: https://leetcode.com/problems/longest-continuous-increasing-subsequence/
 */

class Solution
{
public:
    int findLengthOfLCIS(vector<int> &nums)
    {
        if (nums.empty())
        {
            return 0;
        }

        int thisStart = 0, res = 0;
        int n = nums.size();
        for (int i = 0; i < n - 1; i++)
        {
            if (nums[i + 1] <= nums[i])
            {
                if (i + 1 - thisStart > res)
                {
                    res = i + 1 - thisStart;
                }
                thisStart = i + 1;
            }
        }
        if (n - thisStart > res)
        {
            res = n - thisStart;
        }

        return res;
    }
};